### ISSUE TITLE
IOU-On closing invalid file type modal box user navigated to BNP page

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#38462]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.53-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4429274
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Tap on 1:1 report
3. Tap plus icon --- request money
4. Enter an amount and tap next
5. Tap 3 dots in top & select Add Receipt
6. Upload a password protected pdf file
7. Note in Manual IOU confirmation page, invalid file type modal box shown
8. Tap close
9. Note, user directed to BNP
10. Tap next
11. Note, the closed invalid modal box is shown again.

## Expected Result:
In Manual IOU confirmation page, on closing invalid file type modal box user must remain in same page and must not be navigated to BNP page.

On revisiting Manual IOU confirmation page from BNP, the closed invalid modal box must not be shown again.

## Actual Result:
In Manual IOU confirmation page, on closing invalid file type modal box user is navigated to BNP page.
On revisiting Manual IOU confirmation page from BNP, the closed invalid modal box is shown again.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/4ee13fca-2779-40ef-a075-22dc0496f01c

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The invalid type error message shows again when revisiting the confirmation page.

### What is the root cause of that problem?
When the PDF is detected as a password-protected, it will simply show a modal error message.
https://github.com/Expensify/App/blob/b6c7e12b9f4b1df899ecab9c1f385aa23af1d48e/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L899

When the user presses the modal button, it will simply navigate them to the previous page without clearing the invalid receipt.
https://github.com/Expensify/App/blob/b6c7e12b9f4b1df899ecab9c1f385aa23af1d48e/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L969-L972

So, when they revisit the page, the validation modal error message will show again.

### What changes do you think we should make in order to solve the problem?

We can clear the receipt as soon as the PDF is detected as a password-protected PDF.
```
onPassword={() => {
    setIsAttachmentInvalid(true)
    IOU.setMoneyRequestReceipt(transaction.transactionID, '', '', true);
}}
```

_maybe we can keep the user on page instead of navigating them back_

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
IOU-On closing invalid file type modal box user navigated to BNP page
### What is the root cause of that problem?
We navigate back on pressing `close` button
### What changes do you think we should make in order to solve the problem?
1. Replace [this](https://github.com/Expensify/App/blob/37cde71374e005777c5fa43201ed4d04b8a713d0/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L899C21-L899C68) line with below code
```
onPassword={() => {
    IOU.setMoneyRequestReceipt(transaction.transactionID, "", " ", true);
    setIsAttachmentInvalid(true)
}}
```
2. Update [this](https://github.com/Expensify/App/blob/37cde71374e005777c5fa43201ed4d04b8a713d0/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L971) line like below code
```
onConfirm={isScanRequest ? navigateBack : null}
```

### What alternative solutions did you explore? (Optional)

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
IOU-On closing invalid file type modal box user navigated to BNP page
### What is the root cause of that problem?
We navigate back on pressing `close` button
### What changes do you think we should make in order to solve the problem?
1. Replace [this](https://github.com/Expensify/App/blob/37cde71374e005777c5fa43201ed4d04b8a713d0/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L899C21-L899C68) line with below code
```
onPassword={() => {
    IOU.setMoneyRequestReceipt(transaction.transactionID, "", " ", true);
    setIsAttachmentInvalid(true)
}}
```
2. Update `navigateBack` function like below
```
const navigateBack = () => {
    setIsAttachmentInvalid(false)
    
    if(!isScanRequest){
        Navigation.goBack(ROUTES.MONEY_REQUEST_CREATE_TAB_SCAN.getRoute(iouType, transaction.transactionID, reportID));
    }
}
```
### What alternative solutions did you explore? (Optional)

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
IOU-On closing invalid file type modal box user navigated to BNP page
### What is the root cause of that problem?

### What changes do you think we should make in order to solve the problem?
We always navigate back when we click the `close` button.

### What alternative solutions did you explore? (Optional)
1. Update [this](https://github.com/Expensify/App/blob/37cde71374e005777c5fa43201ed4d04b8a713d0/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L899C21-L899C68) line line below
```
  onPassword={() => {
      IOU.setMoneyRequestReceipt(transaction.transactionID, "", " ", true);
      setIsAttachmentInvalid(true)
  }}
```
2. Update [navigateBack](https://github.com/Expensify/App/blob/37cde71374e005777c5fa43201ed4d04b8a713d0/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L308C1-L310C7) function like below
```
const navigateBack = () => {
    setIsAttachmentInvalid(false)
    if(isScanRequest){
        Navigation.goBack(ROUTES.MONEY_REQUEST_CREATE_TAB_SCAN.getRoute(iouType, transaction.transactionID, reportID));
    }
};
```
But one problem remains: there appears an empty space in place of the image, which disappears after the modal is closed. It causes the UI to bounce. To avoid this behavior, I suggest making the below changes:

3. Add a new variable  (`shouldShowPdf`) to the  components state which is `false` in default and make the style of the [PDFThumbnail](https://github.com/Expensify/App/blob/37cde71374e005777c5fa43201ed4d04b8a713d0/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L896C21-L896C53) component depending on it (`style={shouldShowPdf && styles.moneyRequestImage}`). Also set the `shouldShowPdf` to `true`  in  `onLoadSuccess` method of `PDFThumbnail` component
```
 const [shouldShowPdf, setShouldShowPdf] = useState(false)
 
 ....
 
<PDFThumbnail
    ...
    style={shouldShowPdf && styles.moneyRequestImage}
    onLoadSuccess={()=> setShouldShowPdf(true)}
    ...
/>
```

**Comment by joekaufmanexpensify** [comment]
I agree this is technically a bug, but this is not something we should prioritize fixing. Realistically, there should not be many/any customers attempting to upload a password protected PDF as a receipt. Let's see if we see any real customers actually experience this before doing anything. 

Closing for now.

**Comment by joekaufmanexpensify** [comment]
Actually, I am not even sure this is a bug at all. We're just saying you can't upload a password protected document as a receipt, which feels reasonable. So yeah, I don't think we should do anything about this. 

If a real user reports a case where they want to submit a password protected receipt we can revisit. I am dubious that will ever happen though.

**Comment by joekaufmanexpensify** [comment]
I think it's extremely unlikely this not working exactly correctly will cause issues for any real customers. This is a very edgy edge case. Let's wait and see if anyone actually experiences this before deciding to do anything.

**Comment by joekaufmanexpensify** [comment]
Hmm, can't they get out of that simply by clicking the back `<` in the top left?

**Comment by joekaufmanexpensify** [comment]
Got it. I still don't think this is worth doing anything about now. We're trying to prioritize bugs that are likely to have user impact. I am not sure anyone will really ever experience this. If someone does, we can fix it. But let's wait and see if that happens.
